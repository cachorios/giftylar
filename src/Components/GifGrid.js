import React, { useState } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GiftGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

      const {data: images, loading} = useFetchGifs(category);

      return (
            <>
            <h3 className="animate__animated animate__fadeIn">{ category }</h3>
            { loading && <p className="card-grid animate__animated animate__flash">Cargando</p>}
            <div className="card-grid">
                  { images.map( img => 
                  <GiftGridItem  
                        key = { img.id}
                        {...img} />)}
            </div>
            </>
      )
}
