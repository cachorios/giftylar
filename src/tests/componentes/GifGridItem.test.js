import React from 'react';
import { shallow } from "enzyme"
import { GiftGridItem } from "../../Components/GifGridItem"

describe('Pruebas de <GifGridItem />', () => {
      const title = "El tutulo";
      const url = "http://img.com/elm.jpg";

      test('debe mostrar el componente correctamente', () => {
            
            const wrapper = shallow( <GiftGridItem title={title} url={url}/>);
            expect( wrapper).toMatchSnapshot();
            
      })
      
})
