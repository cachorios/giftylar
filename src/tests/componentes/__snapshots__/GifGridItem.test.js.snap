// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGridItem /> debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="El tutulo"
    src="http://img.com/elm.jpg"
  />
  <p>
    El tutulo
  </p>
</div>
`;
